### PR TITLE
docs(handbook): add icon docs to developer-handbook

### PR DESCRIPTION
**Pull request by joshblack**
Adds some documentation for how to review icon and pictogram pull requests

#### Changelog

**New**

**Changed**

- Add in documentation for how developers can review icon and pictogram pull requests

**Removed**

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
looks good to me

File: docs/developer-handbook.md [hunk 1]
```diff
@@ -627,6 +627,32 @@ You don't need to do anything in `carbon-components` side.
...
+
+_Note: the scaffold task will only apply to the main metadata file, for new icons category information will have to be added by hand_
+
+3. Is the CI check failing because of a merge conflict?
+
+If so, you will need to help out the contributor by resolving the merge conflicts for the Pull Request.
+
+4. Is the CI check failing because of a snapshot change?
+
+If so, you will need to help out the contributor by updating the snapshot based on the changes.
```
Comment (emyarod) [inline docs/developer-handbook.md]: maybe we can add the command here as well? `yarn run test:e2e -u`

**Review by emyarod — COMMENTED** [review]
looks like we need to `prettier` again on this file to pass CI

**Review by dakahn — APPROVED** [review]

File: docs/developer-handbook.md (hunk 1 above)
Comment (joshblack) [reply docs/developer-handbook.md]: Good call! Updated

**Pull request merged**